core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/BindingSetAssignment.java

```diff
@@ -49,13 +50,19 @@
 	}
 
 	@Override
-	public boolean equals(Object other) {
-		return other instanceof BindingSetAssignment;
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null || getClass() != obj.getClass())
```

Comment:
Ref: https://stackoverflow.com/questions/1569688/an-efficient-equalsobject-o-implementation

Choice of .getClass() and instanceOf is relevant if we want to allow for subclassing or not. @jeenbroekstra any preference?

Reply:
I tend to prefer allowing subclasses, so I'd use `!(obj instanceof BindingSetAssignment)` instead.